app/src/main/java/com/kickstarter/libs/KSCurrency.kt

```diff
@@ -0,0 +1,172 @@
+            .currencySymbol(currencyOptions.currencySymbol())
+            .roundingMode(roundingMode)
+            .precision(if (precision > 0) 2 else 0)
+            .build()
+        return NumberUtils.format(currencyOptions.value() ?: 0F, numberOptions).trimAllWhitespace()
+    }
+
+    /**
+     * Returns a boolean determining if a country's currency is ambiguous.
+     * Special case: US people looking at US currency just get the currency symbol.
+     *
+     * @param country The country to check if a code is necessary.
+     * @param excludeCurrencyCode If true, hide the US currency code for US users only.
+     */
+    fun currencyNeedsCode(country: Country, excludeCurrencyCode: Boolean): Boolean {
+        val countryIsUS = country === Country.US
+        val config = currentConfig.observable()
+            .toBlocking()
+            .first()
+        val currencyNeedsCode = config.currencyNeedsCode(country.currencySymbol)
+        val userInUS = config.countryCode() == Country.US.countryCode
+        return if (userInUS && excludeCurrencyCode && countryIsUS) {
+            false
+        } else {
+            currencyNeedsCode
+        }
+    }
+
+    /**
+     * Build [CurrencyOptions] based on the country.
+     */
+    private fun currencyOptions(
+        value: Float,
+        country: Country,
+        excludeCurrencyCode: Boolean
+    ): CurrencyOptions {
+        return CurrencyOptions.builder()
+            .country(country.countryCode)
+            .currencyCode("")
+            .currencySymbol(getCurrencySymbol(country, excludeCurrencyCode))
+            .value(value)
+            .build()
+    }
+
+    /**
+     * Returns the currency symbol for a country.
+     *
+     * @param country The country the currency will be displayed in.
+     * @param excludeCurrencyCode If true, hide the US currency code for US users only.
+     */
+    fun getCurrencySymbol(country: Country, excludeCurrencyCode: Boolean): String {
+        return if (!currencyNeedsCode(country, excludeCurrencyCode)) {
+            country.currencySymbol
+        } else if (country === Country.SG) {
+            // Singapore projects get a special currency prefix
+            "\u00A0" + "S" + country.currencySymbol + "\u00A0"
+        } else if (country.currencySymbol == "kr" || country.currencySymbol == "Fr") {
+            // Kroner projects use the currency code prefix
+            "\u00A0" + country.currencyCode + "\u00A0"
+        } else {
+            "\u00A0" + country.countryCode + country.currencySymbol + "\u00A0"
+        }
+    }
+
+    companion object {
+        /**
+         * Returns a number rounded to the specification.
+         *
+         * @param initialValue Value to convert, local to the project's currency.
+         * @param roundingMode When this is DOWN, we get the floor of the initialValue.
+         */
+        private fun getRoundedValue(initialValue: Double, roundingMode: RoundingMode): Float {
```

Comment:
This method can be private to KSCurrency class, no need for it to be wrapped on a companion object